aeneas/src/ssa/SsaOptimizer.v3

```diff
@@ -835,6 +835,21 @@ class SsaInstrReducer(context: SsaContext) extends SsaInstrMatcher {
 					return replaceOp(i, V3Op.newClassGetMethod(dv));  // ClassGetVirtual[m](K) => ClassGetMethod[m](K)
 				}
 			}
+			VariantGetField(field) => {
+				var receiver = i.input0();
+				if (state.end) return receiver;
+				var unboxed = ClassType.!(receiver.getType()).classDecl.isUnboxed();
+				if (SsaConst.?(receiver)) {
+					// ClassGetField(#K) => #K
+					return graph.valConst(getFieldType(i.op, field),
+						asRecord(receiver).values[field.index]);
+				}
+				i.setFactIf(Fact.O_NO_NULL_CHECK, Fact.O_PURE);
```

Comment:
I think all `VariantGetField` operations are already pure. It's confusing that the `O_NO_NULL_CHECK` fact here basically is for an implementation-level null check--it might even be fully redundant after fixing default-values for variants.